### PR TITLE
improve(FillUtils): Filter SpokePoolClients by update state

### PR DESCRIPTION
**Pull request by pxrl**
This seems to be almost the only change required to support handling SpokePoolClients that fail to complete their update.

### CODE REVIEW

File: src/utils/FillUtils.ts [hunk 1]
```diff
@@ -67,25 +67,25 @@ export async function getUnfilledDeposits(
...
+      const hints = { lowBlock: spokePoolClient.deploymentBlock };
       const startBlock = await getBlockForTimestamp(originChainId, timestamp, blockFinder, redis, hints);
-      originFromBlocks[originChainId] = Math.max(deploymentBlock, startBlock);
+      originFromBlocks[originChainId] = Math.max(spokePoolClient.deploymentBlock, startBlock);
     });
   }
 
   // Iterate over each chainId and check for unfilled deposits.
-  await sdkUtils.mapAsync(chainIds, async (destinationChainId) => {
+  await sdkUtils.mapAsync(chainIds, async (destinationChainId: number) => {
```
Comment (pxrl) [inline src/utils/FillUtils.ts:88]: Driveby change because tsc is complaining that `destinationChainId` implicitly resolves to `any`.

File: src/utils/FillUtils.ts [hunk 2]
```diff
@@ -67,25 +67,27 @@ export async function getUnfilledDeposits(
   depositLookBack?: number
 ): Promise<{ [chainId: number]: RelayerUnfilledDeposit[] }> {
   const unfilledDeposits: { [chainId: number]: RelayerUnfilledDeposit[] } = {};
-  const chainIds = Object.values(spokePoolClients).map(({ chainId }) => chainId);
+  const chainIds = Object.values(spokePoolClients)
+    .filter(({ isUpdated }) => isUpdated)
+    .map(({ chainId }) => chainId);
```
Comment (james-a-morris) [inline src/utils/FillUtils.ts:72]: Should we enforce the numeric type of chainId? It would implicitly type the `chainIds`.
```suggestion
    .map(({ chainId }) => Number(chainId));
```

**Review by james-a-morris — COMMENTED** [review]
One nit

File: src/utils/FillUtils.ts (hunk 2 above)
Comment (pxrl) [reply src/utils/FillUtils.ts:65]: @james-a-morris `SpokePoolClient.chainId` is explicitly defined as a number, so per current implementation it's not necessary and chainIds should be `number[]`. Is the suggestion to protect against `SpokePoolClient.chainId` changing type?

**Review by nicholaspai — APPROVED** [review]
